### ISSUE TITLE
Cannot Add/Edit a product

### ISSUE THREAD

**Issue by AmandaDLyon** [Viveckh/Veniqa#132]
**Describe the bug**
In the management platform, when trying to add or edit a product, the images are successfully uploaded (in the S3 bucket), however, the product does not get added to the database. Refer to the screenshots below to get an understanding of the errors. 

**To Reproduce**
1. Go to the management webclient
2. Try adding a product or editing the text of any existing product
3. Hit Edit 
4. A red notification with pop up stating that the product cannot be edited (not the exact language but you get the idea)

**Expected behavior**

Post the images being successfully uploaded to S3, the products details (JSON) should be successfully added to the product catalogue (database)

**Screenshots**
The errors are the same, whether you are editing an existing product or adding a new one. The ones below were created when I was editing an existing product. 

<img width="1207" alt="Screenshot 2020-07-07 at 2 53 37 PM" src="https://user-images.githubusercontent.com/65491209/87008846-f88cc980-c1e1-11ea-82af-3eb7abecb772.png">
<img width="1123" alt="Screenshot 2020-07-09 at 12 48 16 PM" src="https://user-images.githubusercontent.com/65491209/87009276-92ed0d00-c1e2-11ea-80bc-63a34f10a051.png">
<img width="1070" alt="Screenshot 2020-07-09 at 12 48 28 PM" src="https://user-images.githubusercontent.com/65491209/87009286-95e7fd80-c1e2-11ea-856e-8f2985e62b82.png">
<img width="1115" alt="Screenshot 2020-07-09 at 12 48 48 PM" src="https://user-images.githubusercontent.com/65491209/87009289-97192a80-c1e2-11ea-8cd9-b50bdaf964e7.png">

@Viveckh @smaharj1

**Comment by gfyre** [comment]
@AmandaDLyon I actually had never tried to add or edit a product in through the dashboard. However, I just took a look at it and I am facing the same issue. 

@Viveckh I think that this is quite a pressing issue. Could it be the S3 bucket policies or CORS causing this issue. 

It seems like circular-json is failing or is misconfigured. Can you check to see if the same issue persists on your end?

I am tagging @smaharj1 as well.

**Comment by AmandaDLyon** [comment]
@smaharj1 @Viveckh Any solution or suggestion guys? I am really really looking forward to your help!!

**Comment by AmandaDLyon** [comment]
@smaharj1 @Viveckh this is actually funny. The error exists only when you are adding or editing a product. However, the action that you were trying to do would actually happen.

For example, if you are adding a product, it will show an error, but then the product is actually added to the catalogue. Same goes for editing a product. 

What needs to be done here?

**Comment by AmandaDLyon** [comment]
So now that I am sort of able to upload a product (although that weird-ass issue needs to be fixed), the issue is now on the shopping-client where the images don't load. I get a Code 301, which is basically saying that the image has been Moved Permanently. 

@smaharj1 @Viveckh what can I do here? Somehow, after changing the bucket URL (I am using ap-south-1), it is still able to load the veniqa bucket and not images from my bucket. Where would I change the config for this?

**Comment by smaharj1** [comment]
The issue you are seeing is most probably with communication with S3. Look into bucket policies and who can access the bucket. 

Sorry for not replying earlier. All the founders of this involved in multiple projects. So, this has been kind of in a back burner

**Comment by Viveckh** [comment]
I made a bucket policy update for images and also hosted our demo servers in the same domain as the frontends. With this, the product add/edit seems to function flawlessly in our demo sites. The circular json issue still needs addressing if it persists, but I did not notice that error on the live version during test. 

There's [a new documentation update here](https://documentation.veniqa.com/) if you guys are experiencing issues configuring your S3 bucket for images.
